google/cloud/spanner_v1/pool.py

```diff
@@ -447,6 +521,12 @@ def get(self, timeout=None):
         if timeout is None:
             timeout = self.default_timeout
 
+        start_time = time.time()
+        current_span = get_current_span()
+        current_span.add_event(
+            "Waiting for a session to become available", {"kind": "pinging_pool"}
```

Comment:
Because really it is waiting for a session to become available per
https://github.com/googleapis/python-spanner/blob/b2e67626739c05e8dc2eb81e522ff4931f9fac45/google/cloud/spanner_v1/pool.py#L218-L238
and in our document requirements there is a clear need to add that case
<img width="707" alt="Screenshot 2024-10-05 at 10 43 59 PM" src="https://github.com/user-attachments/assets/54c2a310-5612-414d-99ef-02600263be12">